src/machine/machine_stm32f407.go

```diff
@@ -171,20 +129,24 @@ func (uart UART) Configure(config UARTConfig) {
 		config.BaudRate = 115200
 	}
 
-	// pins
-	switch config.TX {
-	default:
-		// use standard TX/RX pins PA2 and PA3
-		UART_TX_PIN.Configure(PinConfig{Mode: PinModeUartTX})
-		UART_RX_PIN.Configure(PinConfig{Mode: PinModeUartRX})
+	// Set the GPIO pins to defaults if they're not set
+	if config.TX == 0 {
+		config.TX = UART_TX_PIN
+	}
+	if config.RX == 0 {
+		config.RX = UART_RX_PIN
 	}
```

Comment:
Setting defaults for TX and RX pins if not passed in on the `config` object. This is to handle the chip initialization in `runtime/runtime_stm32f407.go`

Reply:
That's a reasonable assumption... could probably also see if the pins equalled each other:
```
if config.TX == config.RX {
...
```
I'll code it up as you recommend to keep the consistency across the project